### ISSUE TITLE
Bots taking teles and skipping path nodes

### ISSUE THREAD

**Issue by EngineerGaming69** [nullworks/cathook#1422]
Bots should take teles shouldn't they? Bots taking teles could be something cool to see in cathook. They could take team tele they see and store is location as a teleport/faster path to that location incase they need to get there quicker.

Skipping path nodes is just something I came up with when editing navmeshes. It could optimize them and even make them look more legit by having them skip a few nav nodes in big spaces instead of walking through each one and wasting time.

I mostly would like to see bots taking teles

**Comment by relaybots** [comment]
The idea of it sounds good, but your overthinking what catbots goals are

They are suppose to kill everyone, and skipping nodes wont help that.

**Comment by EngineerGaming69** [comment]
> The idea of it sounds good, but your overthinking what catbots goals are
> 
> They are suppose to kill everyone, and skipping nodes wont help that.

there not just to kill people. i like having bots that look and act like real people and thats what i use cathook for it.
the node skipping part is just a performance thing that i just thought of, idk if its useful or not.

**Comment by BenCat07** [comment]
teleporters are also dormant, getting the exit position is not easy, We have two options. A: check for a non dormant exit, if close to goal walk on entrace
B: (fallback) check if pitch is in direction of goal

**Comment by BenCat07** [comment]
People will grief bots though.

**Comment by iraizo** [comment]
> People will grief bots though.

Yep, as soon as they will see bots take teleports they will

**Comment by EngineerGaming69** [comment]
it doesnt need to be on it could be toggled in nav so if u have 180 aimbot sniper bots then u wont have it on but if u have legit bots that people dont notice easily u can have teles on

**Comment by TheCrazyInsanity** [comment]
> teleporters are also dormant, getting the exit position is not easy, We have two options. A: check for a non dormant exit, if close to goal walk on entrace
> B: (fallback) check if pitch is in direction of goal

You can save the exit position of the teleporter if the bot ever happens to walk by the exit, Then only take it if it knows the exit. Only problem is if the teleporter gets moved. Maybe you can check for the engineer moving building voiceline and invalidate the coordinates if the bot hears it? Its still terribly ineffective but for the people trying to make human like bots it may be nice.

**Comment by BenCat07** [comment]
Good idea, i can also note the direction of entry and exit, which prevents the engineer from moving the exit unless he manages to by some miracle place it at the exact same angle as the old exit

**Comment by TheCrazyInsanity** [comment]
> Good idea, i can also note the direction of entry and exit, which prevents the engineer from moving the exit unless he manages to by some miracle place it at the exact same angle as the old exit

Didnt think of that. But that would definitely work better than listening for voice commands.
